packages/react-native-renderer/src/ReactFabricHostConfig.js

```diff
@@ -251,10 +249,11 @@ export function createTextInstance(
   hostContext: HostContext,
   internalInstanceHandle: Object,
 ): TextInstance {
-  invariant(
-    hostContext.isInAParentText,
-    'Text strings must be rendered within a <Text> component.',
-  );
+  if (!hostContext.isInAParentText) {
+    if (__DEV__) {
```

Comment:
Can we flip the order of the check so `__DEV__` is first? I'm not sure if GCC is smart enough to know that reading `hostContext.isInAParentText` doesn't have a side-effect and would strip the check, but flipping it will make sure it's stripped from prod builds.

Reply:
Great point! Will do.